core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -40,12 +40,22 @@ trait Queue[F[_], A] { self =>
   /** Dequeue one `A` from this queue. Completes once one is ready. */
   def dequeue1: F[A]
 
+  /** Dequeue all available `A`s from this queue. Completes once any values are ready. */
+  def dequeueAvailable1: F[Vector[A]]
```

Comment:
Updating with the signature `F[NonEmptyChunk[A]]`, guaranteeing that at least one value is returned. We can't hide the chunkiness at the individual effect level - if this is undesired at the API level, we can make it private - but I think it could be useful for imperative-style use.